src/parquet/file/reader-internal.cc

```diff
@@ -254,11 +248,8 @@ void SerializedFile::ParseMetaData() {
   uint8_t footer_buffer[FOOTER_SIZE];
   source_->Seek(filesize - FOOTER_SIZE);
   size_t bytes_read = source_->Read(FOOTER_SIZE, footer_buffer);
-
-  if (bytes_read != FOOTER_SIZE) {
-    throw ParquetException("Invalid parquet file. Corrupt footer.");
-  }
-  if (memcmp(footer_buffer + 4, PARQUET_MAGIC, 4) != 0) {
+  if (bytes_read != FOOTER_SIZE ||
```

Comment:
Hypothetically, since `Read` is a virtual function, it could return fewer than `FOOTER_SIZE` bytes (or, if for some odd reason `fread` failed). It's a pretty pathological issue, though